session_listener.go

```diff
@@ -35,10 +36,17 @@ func sessionListener(addr string) {
 }
 
 func sessionConnectionHandler(conn net.Conn) {
-	conn.SetDeadline(time.Now().Add(messageTimeout))
+	defer conn.Close()
+
+	if err := conn.SetDeadline(time.Now().Add(messageTimeout)); err != nil {
```

Comment:
Added error checking on the deadline stuff, although I don't really expect there to be an error here...